raiden/transfer/channel.py

```diff
@@ -1354,6 +1524,47 @@ def events_for_close(
     return events
 
 
+def events_for_expired_lock(
+        channel_state: NettingChannelState,
+        locked_lock: LockedTransferUnsignedState,
+        pseudo_random_generator: random.Random,
+):
+    nonce = get_next_nonce(channel_state.our_state)
+    locked_amount = get_amount_locked(channel_state.our_state)
+
+    our_balance_proof = channel_state.our_state.balance_proof
+
+    transferred_amount = our_balance_proof.transferred_amount
+    locksroot = EMPTY_MERKLE_ROOT
```

Comment:
this is wrong, removing one lock from the merkle tree doesn't not imply the three will be empty. The locksroot must be computed from the current tree, with the expired lock removed.